llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -3943,11 +3943,13 @@ FixedScalableVFPair LoopVectorizationCostModel::computeFeasibleMaxVF(
   // It is computed by MaxVF * sizeOf(type) * 8, where type is taken from
   // the memory accesses that is most restrictive (involved in the smallest
   // dependence distance).
-  unsigned MaxSafeElements =
-      llvm::bit_floor(Legal->getMaxSafeVectorWidthInBits() / WidestType);
+  unsigned MaxSafeElements = Legal->getMaxSafeVectorWidthInBits() / WidestType;
+  if (Legal->isSafeForAnyVectorWidth())
+    MaxSafeElements = bit_ceil(MaxSafeElements);
+  unsigned MaxSafeElementsPowerOf2 = 1ULL << countr_zero(MaxSafeElements);
```

Comment:
In the case where `isSafeForAnyVectorWidth` returns true doesn't this simply return the same value as `MaxSafeElements`? I just wonder if you can rewrite the code as:

```
  unsigned MaxSafeElementsPowerOf2;
  if (Legal->isSafeForAnyVectorWidth())
    MaxSafeElementsPowerOf2 = bit_ceil(MaxSafeElements);
  else
    MaxSafeElementsPowerOf2 = 1ULL << countr_zero(MaxSafeElements);
```